init.js

```diff
@@ -1,7 +1,29 @@
 'use strict'
 
-const tracer = require('.')
+const path = require('path')
+const Module = require('module')
 
-tracer.init()
+let initBailout = false
 
-module.exports = tracer
+if (process.env.DD_INJECTION_ENABLED) {
+  // If we're running via single-step install, and we're not in the app's
+  // node_modules, then we should not initialize the tracer. This prevents
+  // single-step-installed tracer from clobbering the manually-installed tracer.
+  let resolvedInApp
+  const entrypoint = process.argv[1]
```

Comment:
Would the entry point always be at `argv[1]`? Also, isn't there a way to get the entry point from `Module` and/or `require` anyway?

Reply:
> Would the entry point always be at `argv[1]`

Yes. [The arguments are always normalized like this.](https://nodejs.org/docs/latest/api/process.html#processargv)

> Also, isn't there a way to get the entry point from Module and/or require anyway?

Not at this point in the loading cycle. In a pre-require like this, `require.main` is not set yet, since the value that would point to is only created just prior to loading the entrypoint file.